target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c

```diff
@@ -2942,24 +2946,13 @@ static int airoha_pinctrl_probe(struct platform_device *pdev)
 		return err;
 
 	/* build gpio-chip */
-	err = airoha_pinctrl_add_gpiochip(pinctrl, pdev);
-	if (err)
-		return err;
-
-	return of_platform_populate(dev->of_node, NULL, NULL, dev);
+	return airoha_pinctrl_add_gpiochip(pinctrl, pdev);
 }
 
-static const struct of_device_id of_airoha_pinctrl_match[] = {
```

Comment:
@Ansuel do not drop compatible string even if it is not used at the moment, same for pwm